src/main/java/com/acquia/http/HMACHttpRequestInterceptor.java

```diff
@@ -61,43 +65,65 @@ public HMACHttpRequestInterceptor( String provider, String accessKey, String sec
 
         this.customHeaders = new ArrayList<String>();
     }
-    
+
     /**
      * Sets the custom HTTP header names to use when constructing the message.
      * 
      * @param customHeaders The list of HTTP header names
      */
-    public void setCustomHeaders( String[] customHeaders ) {
-        this.customHeaders = new ArrayList<String>( Arrays.asList( customHeaders ) ) ;
+    public void setCustomHeaders(String[] customHeaders) {
+        this.customHeaders = new ArrayList<String>(Arrays.asList(customHeaders));
     }
-    
+
     /** 
      * Returns the custom header names to use when constructing the message.
      * 
      * @return The list of HTTP header names
      */
     public String[] getCustomHeaders() {
-        return this.customHeaders.toArray(new String[ this.customHeaders.size() ]);
+        return this.customHeaders.toArray(new String[this.customHeaders.size()]);
     }
-    
+
+    /**
+     * Helper method to implode a list of Strings into a String
+     * For example: List [ "a", "b" ] delimiter "," will become "a,b"
+     * @param theList
+     * @param glue
+     * @return
+     */
+    private String implodeStringArray(List<String> theList, String glue) {
+        StringBuilder sBuilder = new StringBuilder();
+        boolean isFirst = true;
+        for (String aString : theList) {
+            if (!isFirst) {
+                sBuilder.append(glue);
+            }
+            sBuilder.append(aString);
+            isFirst = false;
+        }
+        return sBuilder.toString();
+    }
+
     @Override
     public void process(HttpRequest request, HttpContext context) throws HttpException, IOException {
+        StringBuilder authBuilder = HMACUtil.constructAuthorizationString(this.realm,
+            this.accessKey, UUID.randomUUID().toString(), VERSION,
+            this.implodeStringArray(this.customHeaders, ";"), /*signature*/null);
+        request.addHeader("Authorization", authBuilder.toString()); //the info provided so far will need to be encrypted
```

Comment:
Why are you setting the initial http header? If we had an HMACAuthenticationHeader then we could pass that into the HMACMessageCreator#createMessage method